test/test_library/unit/test_database.py

```diff
@@ -0,0 +1,971 @@
+    @unittest.skip('Where is Library.get_id definition?')
+    def test_deletes_dataset_colstats(test):
+        pass
+
+    @unittest.skip('Where is Library.get_id definition?')
+    def test_deletes_dataset(test):
+        pass
+
+    # .delete_dataset_colstats tests
+    def test_deletes_column_stat(self):
+        # prepare state
+        colstat1 = ColumnStatFactory()
+
+        # save id to get rid of ObjectDeletedError.
+        colstat1_d_vid = colstat1.d_vid
+        self.sqlite_db.session.commit()
+
+        # testing.
+        self.sqlite_db.delete_dataset_colstats(colstat1.d_vid)
+        self.assertEquals(
+            self.query(ColumnStat).filter_by(d_vid=colstat1_d_vid).all(),
+            [])
+
+    # .remove_dataset tests
+    def test_removes_dataset_colstats(self):
+        # first assert signatures of the functions we are going to mock did not change.
+        assert_spec(self.sqlite_db.delete_dataset_colstats, ['self', 'dvid'])
+
+        # prepare state.
+        ds1 = DatasetFactory()
+        fake_delete = fudge.Fake().expects_call()
+
+        # testing
+        with fudge.patched_context(LibraryDb, 'delete_dataset_colstats', fake_delete):
+            self.sqlite_db.remove_dataset(ds1.vid)
+        fudge.verify()
+
+    def test_removes_dataset(self):
+
+        # prepare state.
+        ds1 = DatasetFactory()
+        ds1_vid = ds1.vid
+
+        # testing
+        self.sqlite_db.remove_dataset(ds1.vid)
+        self.assertEquals(
+            self.query(Dataset).filter_by(vid=ds1_vid).all(),
+            [],
+            'Dataset was not removed.')
+
+    # .remove_partition_record tests
+    def test_removes_partition_and_stat(self):
+
+        # prepare state
+
+        ds1 = DatasetFactory()
+        table1 = TableFactory(dataset=ds1)
+        partition1 = PartitionFactory(dataset=ds1, table=table1)
+        column1 = ColumnFactory(table=table1)
+        colstat1 = ColumnStatFactory(p_vid=partition1.vid, c_vid=column1.vid, d_vid=ds1.vid)
+        self.sqlite_db.session.commit()
+
+        # save necessary ids for later use.
+        partition_vid = partition1.vid
+        colstat1_c_vid = colstat1.c_vid
+
+        # testing
+        self.sqlite_db.remove_partition_record(partition_vid)
+        partition_query = self.query(Partition).filter_by(vid=partition_vid)
+        self.assertEquals(partition_query.all(), [], 'Partition was not deleted.')
+
+        colstat_query = self.query(ColumnStat).filter_by(c_vid=colstat1_c_vid)
+        self.assertEquals(colstat_query.all(), [], 'ColumnStat instance was not deleted.')
+
+    # .get tests
+    # TODO:
+
+    # .get_table tests
+
+    def test_returns_table(self):
+
+        # prepare state
+        ds1 = DatasetFactory()
+        table1 = TableFactory(dataset=ds1)
+
+        # testing
+        ret = self.sqlite_db.get_table(table1.vid)
+        self.assertIsInstance(ret, Table)
+        self.assertEquals(ret.vid, table1.vid)
+
+    # .tables tests
+    @unittest.skip(
```

Comment:
This may have been updated recently. Library.tables() current implementation is: self.database.session.query(Table).all()